_source/_posts/2022-06-01-jhipster-digital-ocean.md

```diff
@@ -0,0 +1,419 @@
+```
+
+Enable the OIDC authentication in the `jhipster-registry` service by adding the `oauth2` profile in the `kubernetes/registry-k8s/jhipster-registry.yml` file:
+
+```yml
+- name: SPRING_PROFILES_ACTIVE
+  value: prod,k8s,oauth2
+```
+
+## Run locally with Minikube
+
+Install [Minikube](https://minikube.sigs.k8s.io/docs/start/) and [`kubectl`](https://kubernetes.io/docs/tasks/tools/).
+
+For Minkube, you will need at least 2 CPUs. Start MiniKube with your number of CPUs:
+
+```bash
+cd kubernetes
+minikube --cpus <ncpu> start
+```
+Minikube will log the Kubernetes and Docker versions on start:
+
+```
+Preparing Kubernetes v1.23.3 on Docker 20.10.12 ...
+```
+
+For the `store-mongodb` deployment to work, the property `Service.spec.publishNotReadyAddresses` was required, instead of the annotation `service.alpha.kubernetes.io/tolerate-unready-endpoints`, as the latter was deprecated in [Kubernetes release 1.11](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.11.md#kubernetes-111-release-notes).
+
+In the project `kubernetes` folder, edit `store-mongodb.yml` and add the `publishNotReadyAddresses: true` property to the `spec`:
+
+```yaml
+...
+# Headless service for DNS record
+apiVersion: v1
+kind: Service
+metadata:
+  name: store-mongodb
+  namespace: demo
+spec:
+  type: ClusterIP
+  clusterIP: None
+  publishNotReadyAddresses: true
+  ports:
+    - name: peer
+      port: 27017
+  selector:
+    app: store-mongodb
+```
+
+Then deploy the application to Minikube, in the `kubernetes` directory, run:
+
+```bash
+./kubectl-apply.sh -f
+```
+Now is a good time to install [k9s](https://k9scli.io/topics/install/), a terminal-based UI to interact with Kubernetes clusters. Then run k9s with:
+
+```bash
+k9s -n demo
+```
+
+{% img blog/jhipster-digital-ocean/k9s-pods.png alt:"k9s UI" width:"800" %}{: .center-image }
+
+Check out the [commands list](https://k9scli.io/topics/commands/), some useful ones are:
+- `:namespace`: show all available namespaces
+- `:pods`: show all available pods
+You can navigate the pods with `ENTER` and go back with `ESC` keys.
+
+Set up port-forwarding for the JHipster Registry.
+
+```bash
+kubectl port-forward svc/jhipster-registry -n demo 8761
+```
+
+Navigate to `http://localhost:8761` and sign in with your Okta credentials. When the registry shows all services in green, Set up port-forwarding for the gateway as well.
+
+```bash
+kubectl port-forward svc/gateway -n demo 8080
+```
+
+Navigate to `http://localhost:8080`, sign in, and create some entities to verify everything is working fine.
+
+After looking around, stop minikube before the cloud deployment:
+
+```bash
+minikube stop
+```
+
+## Deploy to DigitalOcean Kubernetes
+
+Now that the architecture works locally, let's proceed to the cloud deployment. First, create a [DigitalOcean](https://cloud.digitalocean.com/registrations/new) account, you can try their services with a free $100 credit that is available for 60 days.
```

Comment:
I was asked to pay $5 with PayPal or enter a credit card. I did not see a $100 credit mentioned.

Reply:
I see I did not mention that. I got the 100 credit after paying the $5. I thought that if you enter the credit card you don't have to pay anything. Will update the link to https://www.digitalocean.com/try/free-trial-offer

Reply:
Updated the link and added a comment.